src/bin/cql-stress-cassandra-stress/settings/command/read_write.rs

```diff
@@ -0,0 +1,406 @@
+use crate::settings::{
+    param::{ParamsParser, SimpleParamHandle},
+    ParsePayload,
+};
+use anyhow::{Context, Result};
+use std::str::FromStr;
+use strum_macros::{AsRefStr, EnumString};
+
+use super::{Command, CommandParams};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Uncertainty {
+    pub target_uncertainty: f32,
+    pub min_uncertainty_measurements: u64,
+    pub max_uncertainty_measurements: u64,
+}
+
+impl Uncertainty {
+    pub fn new(
+        target_uncertainty: f32,
+        min_uncertainty_measurements: u64,
+        max_uncertainty_measurements: u64,
+    ) -> Self {
+        Self {
+            target_uncertainty,
+            min_uncertainty_measurements,
+            max_uncertainty_measurements,
+        }
+    }
+
+    pub fn print_settings(&self) {
+        println!("  Target Uncertainty: {}", self.target_uncertainty);
+        println!(
+            "  Minimum Uncertainty Measurements: {}",
+            self.min_uncertainty_measurements
+        );
+        println!(
+            "  Maximum Uncertainty Measurements: {}",
+            self.max_uncertainty_measurements
+        );
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, AsRefStr, EnumString)]
+#[strum(serialize_all = "SCREAMING_SNAKE_CASE")]
+#[strum(ascii_case_insensitive)]
+pub enum Truncate {
+    Never,
+    Once,
+    Always,
+}
+
+impl Truncate {
+    fn parse(truncate: &str) -> Result<Self> {
+        Self::from_str(truncate).context(format!("Invalid truncate type: {}", truncate))
+    }
+
+    fn show(&self) -> &str {
+        self.as_ref()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, AsRefStr, EnumString)]
+#[strum(serialize_all = "SCREAMING_SNAKE_CASE")]
+#[strum(ascii_case_insensitive)]
+pub enum ConsistencyLevel {
+    One,
+    Quorum,
+    LocalQuorum,
+    EachQuorum,
+    All,
+    Any,
+    Two,
+    Three,
+    LocalOne,
+    Serial,
+    LocalSerial,
+}
+
+impl ConsistencyLevel {
+    fn parse(cl: &str) -> Result<Self> {
+        Self::from_str(cl).context(format!("Invalid consistency level: {}", cl))
```

Comment:
The `format!` here always allocates a string (because of `format!`). It's better to use `with_context` if you need to perform expensive computation on error:

```rust
Self::from_str(cl).with_context(|| format!("Invalid consistency level: {}", cl))
```